plugin/src/main/kotlin/com/moengage/internal/MavenPublishManager.kt

```diff
@@ -47,15 +48,29 @@ internal class MavenPublishManager(private val project: Project) {
         releaseVariant = project.findProperty(RELEASE_VARIANT) as? String ?: DEFAULT_RELEASE_VARIANT
         releasePortal = ArtifactReleasePortal.getMavenCentralPortal(project.findProperty(RELEASE_HOST) as? String)
 
-        val serviceBuilder =
-            ServiceBuilder(releasePortal, project.getUserName(releasePortal), project.getUserPassword(releasePortal))
+        val maxRetryCountOnTimeOut =
```

Comment:
Should there be a limit on the maximum retry? like, say 5 or 10? Beyond that would be DDOS on the server if the server down or something

Reply:
Added max retry as 5